### PR TITLE
Adds support for `@included` in compaction, expansion and flattening.

### PR DESCRIPTION
**Pull request by gkellogg**
Code added for framing, but not enabled due to big backlog on framing in general.

### CODE REVIEW

**Comment by gkellogg** [comment]
Not directly related, but it touched on some of the same code, so I added some support for keyword oddities.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,8 +2,14 @@
 
 ## 2.0.1 - 2019-12-10
 
+### Added
```
Comment (davidlehn) [inline CHANGELOG.md]: Move all the changelog updates above 2.0.1.

File: lib/context.js [hunk 2]
```diff
@@ -350,6 +350,11 @@ api.createTermDefinition = ({
       'Invalid JSON-LD syntax; keywords cannot be overridden.',
       'jsonld.SyntaxError',
       {code: 'keyword redefinition', context: localCtx, term});
+  } else if(term.match(/@[a-zA-Z]+$/)) {
```
Comment (davidlehn) [inline lib/context.js]: Just want to find things that start with `@...` right?
```suggestion
  } else if(term.match(/^@[a-zA-Z]+$/)) {
```

Comment (davidlehn) [reply lib/context.js]: Same with other patterns below.

Comment (davidlehn) [reply lib/context.js]: Might be useful to move the regex to a common const at the top of the file.

File: lib/context.js [hunk 3]
```diff
@@ -350,6 +350,11 @@ api.createTermDefinition = ({
...
+    // FIXME: remove logging and use a handler
+    console.warn('WARNING: terms beginning with "@" are reserved' +
```
Comment (davidlehn) [inline lib/context.js]: Using `console.warn` here is the same issue that resulted in the `protectedMode` `warn/error` flag.  That was supposed to eventually change a better handler, but that hasn't been designed yet.  Might want another similar flag and we can fix them all at once.  The machinery to pass around those options still exists so would be easy to add.  I'm not sure what the best behavior even is.

Comment (gkellogg) [reply lib/context.js]: I'll leave this for you to come up with a comprehensive solution. There are other areas where the spec says to generate warnings.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,9 +1,14 @@
 # jsonld ChangeLog
 
-## 2.0.1 - 2019-12-10
+### Added
+- Added support for `@included` blocks
+- Skip things that have the form of a keyword, with warning.
 
 ### Fixed
 - JSON literal value handling issues.
```
Comment (davidlehn) [inline CHANGELOG.md]: This one line should be for 2.0.1.

File: lib/context.js (hunk 3 above)
Comment (davidlehn) [reply lib/context.js]: Ok.  Printing out warnings to a console can be invisible for backend code and UIs, hence the idea to have some of handler.  I haven't put time into a design for that.  A more strict mode that throws errors seems useful to me as well.

File: lib/context.js [hunk 5]
```diff
@@ -437,6 +443,15 @@ api.createTermDefinition = ({
         'absolute IRI or a blank node identifier.',
         'jsonld.SyntaxError', {code: 'invalid IRI mapping', context: localCtx});
     }
+
+    if(reverse.match(KEYWORD_PATTERN)) {
+      // FIXME: remove logging and use a handler
+      console.warn('WARNING: values beginning with "@" are reserved' +
+        ' for future use and ignored', {reverse});
+      activeCtx.mappings.delete(term);
```
Comment (davidlehn) [inline lib/context.js]: In the case of warnings (here and below), should this restore the `previousMapping` for `term` instead of deleting it?

Comment (gkellogg) [reply lib/context.js]: Yes, it should restore before returning; I'll take care of that.

Comment (gkellogg) [reply lib/context.js]: Fixed in 649857c33c42774640b51b83ed1e294f7d5a2b92.

**Pull request merged**